django/db/backends/base/schema.py

```diff
@@ -864,7 +864,8 @@ def _alter_field(
         # Drop any FK constraints, we'll remake them later
         fks_dropped = set()
         if (
-            self.connection.features.supports_foreign_keys
+            self._fk_constraints_should_be_recreated()
+            and self.connection.features.supports_foreign_keys
```

Comment:
```suggestion
            self.connection.features.supports_foreign_keys
            and self.connection.features.recreate_foreign_key_constraints
```

As @shangxiao mentioned, this should be implemented using a feature flag. See `django/db/backends/base/features.py`.